From f34d2a286aa3fa5e7dcd3db4d507c087331c0730 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Fri, 11 Aug 2017 18:54:59 +0800
Subject: [PATCH 2379/5242] MLK-16197-4 video: fbdev: dcss: improve hp/lp data
 count settings

commit  d4cc356d0dc74d9b808ced30b4ea186d29e152ff from
https://source.codeaurora.org/external/imx/linux-imx.git

For now, all the DCSS register configuration should be put
in the high priority single buffer by default in context
loader. So improve the high and low priority data counts
calculation and setting.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index 184f3c3..bca7517 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -2406,7 +2406,7 @@ static void dcss_ctxld_config(struct work_struct *work)
 		writel(cfifo->dma_handle + cc->fifo_in * kfifo_esize(&cfifo->fifo),
 		       info->base + chans->ctxld_addr + CTXLD_SB_BASE_ADDR);
 		writel(cc->sb_hp_data_len |
-		       (cc->sb_data_len - cc->sb_hp_data_len),
+		       ((cc->sb_data_len - cc->sb_hp_data_len) << 16),
 		       info->base + chans->ctxld_addr + CTXLD_SB_COUNT);
 	}
 
@@ -2490,6 +2490,7 @@ static int commit_to_fifo(uint32_t channel,
 			count = kfifo_out(&cfifo->fifo, cb->sb_addr, count);
 			BUG_ON(1);
 		}
+		cc->sb_hp_data_len = count;
 		cc->sb_data_len = count;
 	}
 
-- 
1.7.9.5

